packages/engine/lib/memory/src/shared_memory/segment.rs

```diff
@@ -83,16 +83,21 @@ impl<'id> MemoryId<'id> {
 
     /// Clean up generated shared memory segments associated with a given `MemoryId`.
     pub fn clean_up<Id: Borrow<Uuid>>(id: Id) {
+        // We're ignoring glob errors as they shouldn't stop the whole cleanup process.
```

Comment:
I think we should have this function return a `Result` and let the function that called it ignore the error. 
An Error from `glob::glob` will stop the clean-up of **all** the shared memory files here.

I agree that generally the whole clean-up step (not just this function) shouldn't fail early, but this function should report when glob fails.

```rust
let shm_files = glob::glob(&format!("/dev/shm/{}_*", Self::prefix(id)))?;
```

Reply:
Applied in https://github.com/hashintel/hash/pull/494/commits/1d06d0fb07540c2514eb04225c48d2e516a37f73